userguide/storage.rst

```diff
@@ -713,53 +713,70 @@ existing datasets to close them--the remaining
+enabling dedup is **usually not recommended** and the setting should be left in
+the default setting ("off") in most situations.
+
+.. warning::
+
+   The more data written to a deduplicated dataset, the more RAM it 
+   requires. The DDTs (dedup tables) also compete with other data for RAM,
+   and can be pushed out of RAM in some cases. When the system starts 
+   storing DDTs on disk because they no longer fit into RAM or have been
+   pushed out of RAM, performance craters.
+   
+   Importing an unclean deduplicated pool can also require large amounts of RAM, 
+   so that the deduplicated data in the pool can be checked during import. 
+   If the system does not have the needed RAM, it will panic. The only 
+   solution then is to add more RAM or recreate the pool.
+
+   **There is also no quick way to undedup data within a dataset once deduplication 
+   is enabled, since disabling deduplication DOES NOT CHANGE existing data.** The 
+   simplest way to undedup existing data is to copy the deduped data to a new dataset 
+   or location that does not have dedup set. The data will be undeduped when written. 
+   The simplest way to undedup data and keep it in the same pool, zvol or dataset is 
+   as follows: disable dedup on the pool, zvol or dataset, then copy the data within 
+   the same pool, zvol or dataset. Once the copy is confirmed to be correct, the 
+   original data can optionally be deleted. If the pool does not have enough free space
+   for this procedure, then disable dedup, copy the data to another system or storage
+   device, check the copies are correct and safe, then delete the original data, and 
+   transfer back the copied data to the pool.
+
+   For these reasons, **think carefully before enabling dedup!** 
+   Deduplication is **usually not recommended unless the data contains a 
+   very substantial level of duplication, and the system (especially 
+   RAM) can cope acceptably with the additional burden**. In most cases, 
+   compression provides storage gains comparable to deduplication with 
+   less impact on performance.  This `article 
+   <http://constantin.glez.de/blog/2011/07/zfs-dedupe-or-not-dedupe>`_ 
+   provides a good description of the value versus cost considerations 
+   for deduplication. Unless a lot of RAM and a lot of highly duplicate 
+   data is available, **do not change the default deduplication setting 
+   of "Off"!**  
+
+In %brand%, deduplication can be enabled during dataset creation, and 
+occurs at the time when files are read and written. The setting can 
+be changed at any time but will not affect data already written to 
+the pool.  If the deduplication method is set to *Verify*, ZFS will 
+do a byte-to-byte comparison when two blocks have the same hash signature 
+to make sure that the block contents are indeed identical. Since hash 
+collisions are extremely unblikely, *Verify* is usually not worth the 
+performance hit.
```

Comment:
Is this repeated information? I'd say combine this information into the sections above where applicable. After that turn only what's different should be in this final paragraph.

Reply:
This para covers 2 distinct points: 

1)  dedup is set at creation and can be modified afterwards. (a general point about how it works on *NAS)
2) the dedup algorithm can be changed and verfiy used, but isn't that helpful normally. (a specific choice of parameter)

I've split these and moved to better places, merging in as possible. The first is moved right near the start of the section, the second is left further down.

Reply:
This para covers 2 distinct points: 

1)  dedup is set at creation and can be modified afterwards. (a general point about how it works on *NAS)
2) the dedup algorithm can be changed and verfiy used, but isn't that helpful normally. (a specific choice of parameter)

I've split these and moved to better places, merging in as possible. The first is moved right near the start of the section, the second is left further down.

I notice that some of the info is omitted for TrueNAS but other info on dedup is left in. Can you check if the info left in/out is correct?

Reply:
This para covers 2 distinct points: 

1)  dedup is set at creation and can be modified afterwards. (a general point about how it works on *NAS)
2) the dedup algorithm can be changed and verfiy used, but isn't that helpful normally. (a specific choice of parameter)

I've split these and moved to better places, merging in as possible. The first is moved right near the start of the section, the second is left further down.

I notice that some of the info is omitted for TrueNAS but other info on dedup is left in. Can you check if the info left in/out is correct?